httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CacheControlHeaderParser.java

```diff
@@ -113,28 +113,14 @@ protected CacheControlHeaderParser() {
      * <p>
      * "s-maxage" (-1).</p>
      *
-     * @param header the header to parse, cannot be {@code null}
+     * @param headers the header to parse, cannot be {@code null}
      * @return a new {@link CacheControl} instance containing the relevant caching directives parsed from the header
      * @throws IllegalArgumentException if the input header is {@code null}
      */
-    public final CacheControl parse(final Header header) {
-        Args.notNull(header, "Header");
-
-        final CharArrayBuffer buffer;
-        final Tokenizer.Cursor cursor;
-        if (header instanceof FormattedHeader) {
-            buffer = ((FormattedHeader) header).getBuffer();
-            cursor = new Tokenizer.Cursor(((FormattedHeader) header).getValuePos(), buffer.length());
-        } else {
-            final String s = header.getValue();
-            if (s == null) {
-                return new CacheControl();
-            }
-            buffer = new CharArrayBuffer(s.length());
-            buffer.append(s);
-            cursor = new Tokenizer.Cursor(0, buffer.length());
-        }
+    public final CacheControl parse(final Header[] headers) {
```

Comment:
@arturobernalg Please use `Iterator<Header>` instead. One can then use `Iterator<Header> MessageHeaders#headerIterator(String name)` without creating an intermediate array of `Header`s.